src/CaptureClient/ApiEventIdProvider.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureClient/ApiEventIdProvider.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <memory>
+
+namespace orbit_capture_client {
+
+[[nodiscard]] std::unique_ptr<NameEqualityApiEventIdProvider>
+NameEqualityApiEventIdProvider::Create(const orbit_grpc_protos::CaptureOptions& capture_options) {
+  const auto& instrumented_functions = capture_options.instrumented_functions();
+  const uint64_t max_id =
+      instrumented_functions.empty()
+          ? 0
+          : std::max_element(
+                std::begin(instrumented_functions), std::end(instrumented_functions),
+                [](const auto& a, const auto& b) { return a.function_id() < b.function_id(); })
+                ->function_id();
+  return std::make_unique<NameEqualityApiEventIdProvider>(max_id + 1);
+}
+
+NameEqualityApiEventIdProvider::NameEqualityApiEventIdProvider(uint64_t start_id)
+    : next_id_(start_id) {}
+
+[[nodiscard]] uint64_t NameEqualityApiEventIdProvider::ProvideId(const TimerInfo& timer_info) {
+  const std::string& name = timer_info.api_scope_name();
```

Comment:
This is ("correctly") using the name of the track for async scopes, right?

Reply:
Is it? I think api_scope_name for async scopes is the track name.....

Reply:
You would get the Scope Name from `ManualInstrumentationManager::GetString`. But I think we should not go this way. I'd suggest to not add Async events in this PR.

Reply:
> I think api_scope_name for async scopes is the track name

Yeah yeah, that's what I said.

We haven't really discussed whether using the track name is what we want, but at this point it already looks to me like the only reasonable option (hence my "correctly" in quotes). Using the string in the async slice itself would be problematic as that's optional and very likely to be different for every async scope. `Orbit.h`:

```
// Example usage: Tracking "File IO" operations.
// Thread 1: ORBIT_START_ASYNC("File IO", unique_64_bit_id);  // File IO request site.
// Thread 1 or 2: ORBIT_ASYNC_STRING(unique_64_bit_id, "My very long file path");
// Thread 1 or 2: ORBIT_STOP_ASYNC(unique_64_bit_id);  // File IO result site.
// Result: Multiple time slices labeled with the results of "io_request->GetFileName()" will appear
//         on a single "async" track named "File IO".
```

Reply:
As also discussed offline, this is intended -- we will use track name for grouping.